pkg/proxy/util/utils.go

```diff
@@ -288,7 +288,14 @@ func MapCIDRsByIPFamily(cidrStrings []string) map[v1.IPFamily][]string {
 		if ipFamily, err := getIPFamilyFromCIDR(cidr); err == nil {
 			ipFamilyMap[ipFamily] = append(ipFamilyMap[ipFamily], cidr)
 		} else {
-			klog.ErrorS(nil, "Skipping invalid CIDR", "cidr", cidr)
+			// this function is called in multiple places. All of which
+			// have sanitized data. Except the case of ExternalIPs which is
+			// not validated by api-server. Specifically empty strings
+			// validation. Which yields into a lot of bad error logs.
+			// check for empty string
+			if cidr != "" && cidr != " " {
+				klog.ErrorS(nil, "Skipping invalid CIDR", "cidr", cidr)
```

Comment:
the bug is in the previous function
```
Skipping invalid IP: ".
```

this only used for nodePortAddresses and loadBalancerSourceRanges
```
pkg/proxy/iptables/proxier.go:  ipFamilyMap := utilproxy.MapCIDRsByIPFamily(nodePortAddresses)
pkg/proxy/iptables/proxier.go:  ipFamilyMap := utilproxy.MapCIDRsByIPFamily(nodePortAddresses)
pkg/proxy/ipvs/proxier.go:      ipFamilyMap := utilproxy.MapCIDRsByIPFamily(nodePortAddresses)
pkg/proxy/ipvs/proxier.go:      ipFamilyMap := utilproxy.MapCIDRsByIPFamily(nodePortAddresses)
pkg/proxy/util/utils_test.go:func TestMapCIDRsByIPFamily(t *testing.T) {
pkg/proxy/util/utils_test.go:                   cidrMap := MapCIDRsByIPFamily(testcase.ipString)
pkg/proxy/util/utils.go:// MapCIDRsByIPFamily maps a slice of IPs to their respective IP families (v4 or v6)
pkg/proxy/util/utils.go:func MapCIDRsByIPFamily(cidrStrings []string) map[v1.IPFamily][]string {
pkg/proxy/service.go:   ipFamilyMap = utilproxy.MapCIDRsByIPFamily(loadBalancerSourceRanges)
```

loadbalancerSourceRanges had the same issue and was fixed with this https://github.com/kubernetes/kubernetes/pull/94107